neo/SmartContract/ApplicationEngine.cs

```diff
@@ -20,7 +20,7 @@ public partial class ApplicationEngine : ExecutionEngine
         private readonly List<IDisposable> disposables = new List<IDisposable>();
 
         public TriggerType Trigger { get; }
-        public IVerifiable ScriptContainer { get; }
+        public ConsumableWitnesses ScriptContainer { get; }
```

Comment:
I think that a better approach here, is having a vector of verified scripthashes. Lets keep only IVerifiable class,and we remove consumed witnesses from that list.
So we load the script and already pass the available validated witnesses.